platform/admin/core/admin-core-alertservice/src/test/groovy/org/codice/ddf/admin/core/alert/service/internal/AlertServiceBeanTest.groovy

```diff
@@ -0,0 +1,264 @@
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.admin.core.alert.service.internal
+
+import org.codice.ddf.admin.core.alert.service.data.internal.Alert
+import org.osgi.service.event.Event
+import spock.lang.Specification
+
+import javax.management.MalformedObjectNameException
+import javax.management.ObjectName
+import java.lang.management.ManagementFactory
+
+class AlertServiceBeanTest extends Specification {
+
+    final mbeanName = new ObjectName(AlertServiceBean.class.getName() + ":service=alert-service")
+
+    def alertServiceBean
+
+    void setup() throws Exception {
+        // the keystore path must be set for the InsecureDefaultsService to use
+        System.setProperty("javax.net.ssl.keyStore", "TestKeystorePath")
+        alertServiceBean = new AlertServiceBean(ManagementFactory.getPlatformMBeanServer())
+        assert !isRegistered()
+        assert alertServiceBean.addedAlerts.isEmpty()
+    }
+
+    void cleanup() throws Exception {
+        if (isRegistered()) {
+            ManagementFactory.getPlatformMBeanServer()
+                    .unregisterMBean(mbeanName)
+        }
+    }
+
+    def 'test init'() throws Exception {
+        when:
+            alertServiceBean.init()
+
+        then:
+            isRegistered()
+    }
+
+    def 'test init when the bean has already been registered'() throws Exception {
+        when:
+            alertServiceBean.init()
+
+        then:
+            isRegistered()
+
+        when:
+            alertServiceBean.init()
+
+        then:
+            isRegistered()
+    }
+
+    def 'test destroy'() throws Exception {
+        when:
+            alertServiceBean.init()
+
+        then:
+            isRegistered()
+
+        when:
+            alertServiceBean.destroy()
+
+        then:
+            !isRegistered()
+    }
+
+    def 'test destroy when the bean has not already been registered'() throws Exception {
+        when:
+            alertServiceBean.destroy()
+
+        then:
+            !isRegistered()
+    }
+
+    private boolean isRegistered() throws MalformedObjectNameException {
+        return ManagementFactory.getPlatformMBeanServer()
+                .isRegistered(mbeanName)
+    }
+
+    def 'test getAlerts with the default InsecureDefaultsService alerts'() {
+        when:
+            final alerts = alertServiceBean.getAlerts()
+
+        then:
+            alerts.size() == 1
+            verifyInsecureDefaultsServiceDefaultAlerts(alerts)
+    }
+
+    def 'test handling an invalid alert event'(event) {
+        when:
+            alertServiceBean.handleEvent(event)
+
+        then:
+            alertServiceBean.addedAlerts.isEmpty()
+
+        when:
```

Comment:
❗️ As a general rule, tests shouldn't repeat the `when` and `then` blocks. When they do, it usually means that one of the pairs can be extracted to a separate test. In this case, I don't think we need to put the `getAlerts` in a `when` since it's really just code used to test the state of the object, not the operation being tested. See http://spockframework.org/spock/docs/1.0/spock_primer.htm for more on this.